src/thinkspace/api/thinkspace-casespace/app/models/thinkspace/casespace/assignment.rb

```diff
@@ -235,6 +235,12 @@ def delete_ownerable_data(ownerables)
         end
       end
 
+      def sync_rat
+        return false unless settings.present?
+        return false unless settings['rat']['sync'].present?
```

Comment:
Concerned about this line.  I could see a `undefined method [] on nil` type of error if settings['rat'] was undefined.  Consider using a dig instead: `settings.dig('rat', 'sync')`.

This could be `return false unless settings.present? && settings.dig('rat', 'sync').present?`